typescript/infra/src/agents/aws/validator.ts

```diff
@@ -50,6 +50,16 @@ export class S3Validator extends BaseValidator {
     this.s3Bucket = new S3Wrapper(s3Bucket);
   }
 
+  async getLatestCheckpointIndex() {
+    const latestCheckpointIndex = await this.s3Bucket.getS3Obj<number>(
+      LATEST_KEY,
+    );
+
+    if (!latestCheckpointIndex) return -1;
```

Comment:
this could be misleading

Reply:
What do you think should happen instead? I wanted to just surface errors in fetching this by -1